src/components/strategies/create/useDuplicateStrategy.ts

```diff
@@ -1,91 +1,96 @@
-import { useNavigate, useSearch, StrategyCreateSearch } from 'libs/routing';
-import { Strategy } from 'libs/queries';
-import { isOverlappingStrategy } from '../overlapping/utils';
+import { useSearch, StrategyCreateSearch } from 'libs/routing';
+import { Order, Strategy } from 'libs/queries';
+import { isOverlappingStrategy } from 'components/strategies/overlapping/utils';
+import { useTokens } from 'hooks/useTokens';
 
-const isValid = (strategy: Strategy) => {
-  return (
-    (strategy.hasOwnProperty('base') && strategy.hasOwnProperty('quote')) ||
-    (strategy.hasOwnProperty('order0') && strategy.hasOwnProperty('order1'))
-  );
+const isEmptyOrder = (order: Order) => {
+  return !Number(order.startRate) && !Number(order.endRate);
 };
-
-export const toStrategyCreateSearch = (
-  strategy: Strategy
-): StrategyCreateSearch => {
-  const { order0, order1 } = strategy;
-  const isRecurring = order0.endRate !== '0' && order1.endRate !== '0';
-  const isBuyLimit = order0.startRate === order0.endRate;
-  const isSellLimit = order1.startRate === order1.endRate;
-  const isLimit = isBuyLimit && isSellLimit;
-  if (isRecurring) {
-    return {
-      strategyType: 'recurring',
-      strategySettings: isLimit ? 'limit' : 'range',
-    };
-  } else {
-    return {
-      strategyType: 'disposable',
-      strategySettings: isLimit ? 'limit' : 'range',
-      strategyDirection: order1.endRate === '0' ? 'buy' : 'sell',
-    };
-  }
+const isLimitOrder = (order: Order) => {
+  return order.startRate === order.endRate;
 };
 
-const decodeStrategyAndValidate = (
-  urlStrategy?: string
-): (Strategy & StrategyCreateSearch) | undefined => {
-  if (!urlStrategy) return;
-
-  try {
-    const decodedStrategy = JSON.parse(atob(urlStrategy));
-    if (!isValid(decodedStrategy)) return;
-    return {
-      ...decodedStrategy,
-      ...toStrategyCreateSearch(decodedStrategy),
-    };
-  } catch (error) {
-    console.log('Invalid value for search param `strategy`', error);
+/** Round to 6 decimals after leading zeros */
+const roundSearchParam = (param: string) => {
```

Comment:
lets expose this in utils, I will have to utilise this in the simulator as well. And I would probably do this directly in the `validateSearch` option in the router.